packages/core-data/src/hooks/use-entity-record.ts

```diff
@@ -57,10 +62,10 @@ interface Options {
  * application, the page and the resolution details will be retrieved from
  * the store state using `getEntityRecord()`, or resolved if missing.
  *
- * @return {EntityRecordResolution<RecordType>} Entity record data.
+ * @return Entity record data.
  * @template RecordType
  */
-export default function __experimentalUseEntityRecord< RecordType >(
+export default function useEntityRecord< RecordType >(
```

Comment:
should we not leave the `__experimentalUseEntityRecord` in place for a bit but add a `deprecation()` to it so people get some warning to switch over if they are using it?

Reply:
Good question, but isn't the purpose of __experimental to not require deprecation?

Reply:
Is that? I know part of the intention is to lead people to avoid using those but I think the reality is that many do.

It's not my call and I don't really know our polity (maybe a good update to the docs is in store) but personally I think we can avoid a lot of harm and loss of trust if we do give a due period of deprecation for the core APIs. If after six months we remove something that's been deprecated then at least we gave notice and opportunity to update code. Things move slow even when things move fast.

Reply:
The idea with `__experimental` APIs is to not have people using them outside of the Gutenberg repository until they stabilize:

https://developer.wordpress.org/block-editor/contributors/code/coding-guidelines/#experimental-and-unstable-apis

> An experimental or unstable API is named as such to communicate instability of a function whose interface is not yet finalized. Aside from references within the code, these APIs should neither be documented nor mentioned in any CHANGELOG. They should effectively be considered to not exist from an external perspective. In most cases, they should only be exposed to satisfy requirements between packages maintained in this repository.

There's a related discussion going on in https://github.com/WordPress/gutenberg/issues/40316

Technically it wouldn't be too hard to add an additional `__experimental` export that calls the underlying function with a deprecation notice, but it is not how it usually happens. I'd move forward here as is, and then have the deprecation policy discussion separately.

Reply:
I guess it won't harm to add the deprecation notice for the old method name and remove it with WordPress 6.1 or 6.2.